Lib/cgi.py

```diff
@@ -605,7 +609,7 @@ def read_multi(self, environ, keep_blank_values, strict_parsing):
             query = urllib.parse.parse_qsl(
                 self.qs_on_post, self.keep_blank_values, self.strict_parsing,
                 encoding=self.encoding, errors=self.errors,
-                max_num_fields=self.max_num_fields)
+                max_num_fields=self.max_num_fields, separator=separator)
```

Comment:
Ditto. It seems that some of the arguments for ``read_multi`` are instead passed directly to ``klass`` (below), whereas all the ``parse_qsl`` stuff uses settings from ``self.``
```suggestion
                max_num_fields=self.max_num_fields, separator=self.separator)
```